tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Steps/UserSteps.cs

```diff
@@ -7,11 +17,82 @@ namespace NHSD.BuyingCatalogue.Identity.Api.IntegrationTests.Steps
     [Binding]
     internal sealed class UserSteps
     {
+        private readonly string _organisationUrl;
+
         private readonly ScenarioContext _context;
+        private readonly Response _response;
+        private readonly Settings _settings;
+        private readonly ContextConstants _contextConstants;
 
-        public UserSteps(ScenarioContext context)
+        public UserSteps(ScenarioContext context, Response response, Settings settings)
         {
             _context = context;
+            _response = response;
+            _settings = settings;
+            _organisationUrl = settings.OrganisationApiBaseUrl + "/api/v1/Organisations";
+            _contextConstants = new ContextConstants();
+        }
+
+        [Given(@"Users exist")]
+        public async Task GivenUsersExist(Table table)
+        {
+            var users = table.CreateSet<NewUserTable>();
+            foreach (var user in users)
+            {
+                var allOrganisations = _context.Get<IDictionary<string, Guid>>(_contextConstants.OrganisationMapDictionary);
+
+                var organisationId = Guid.Empty;
+                if (allOrganisations.ContainsKey(user.OrganisationName))
+                {
+                    organisationId = allOrganisations[user.OrganisationName];
+                }
+
+                var userEntity =
+                    new UserEntity
+                    {
+                        PasswordHash = GenerateHash(user.Password),
+                        FirstName = user.FirstName,
+                        LastName = user.LastName,
+                        Email = user.Email,
+                        Disabled = user.Disabled,
+                        PhoneNumber = user.PhoneNumber,
+                        Id = user.Id,
+                        OrganisationId = organisationId,
+                        OrganisationFunction = "TestUser"
```

Comment:
Hmm. I'm currently using OrganisationFunction as a means of clearing all entities before a run. It could be switched to something else though that would serve the same purpose and leave OrganisationFunction free to be Authority/Buyer